### ISSUE TITLE
How to implement extended "oauthlib.openid.RequestValidator" as explained in "openID Connect" section of oauthlib docs.

### ISSUE THREAD

**Issue by Wrufesh** [jazzband/django-oauth-toolkit#742]
I was trying to implement OpenId Connect with `django-oauth-toolkit`.
I was not able to figure out how to achieve that following the docs.

I believe django-oauth-toolkit is wrapper of oauthlib, we must be able to do so as oauthlib supports that.
I found this [oauthlib docs](https://oauthlib.readthedocs.io/en/latest/oauth2/oidc/validator.html#requestvalidator-extension).
Here it is described how to implement openid connect with oauthlib.

How can I reflect it in django-oauth-toolkit as it does not provide an interface to override "OAUTH2_VALIDATOR_CLASS" and "OAUTH2_SERVER_CLASS" settings.
Thanks

**Comment by IvanAnishchuk** [comment]
Well, yes and no. Yes, you can probably provide the right request validator and server classes (and probably most of the models) and most of the stuff should work but request validator will probably require significant changes and changing some other code might be required as well. I believe DOT's request validator currently has quite a few things hardcoded to use bearer tokens and oauth2 flows though I didn't specifically review it regarding this.

More importantly, note that openid connect provides specs for more than just issuing tokens and controlling access. Everything else, like, user info management api, for example, will need to be implemented somewhere. Generally speaking, I agree that we should add better support for JWT tokens, openid flows, metadata, etc. in DOT where appropriate, would love to review PRs on the subject.

**Comment by dopry** [comment]
Yes you should be able to override the server or validator.. but you can also set OIDC_ENABLE these days

https://django-oauth-toolkit.readthedocs.io/en/latest/oidc.html
